packages/cli/lib/services/deploy.service.ts

```diff
@@ -35,7 +35,12 @@ class DeployService {
             printDebug(`Environment is set to ${environmentName}`);
         }
 
-        await compileAllFiles({ debug });
+        const successfulCompile = await compileAllFiles({ debug });
+
+        if (!successfulCompile) {
+            console.log(chalk.red('Compilation was not fully successful. Please make sure all files compile before deploying'));
+            return;
```

Comment:
this kind of errors should probably process.exit(1) otherwise it won't break the CI